assert/assertions_test.go

```diff
@@ -6121,8 +6164,54 @@ func TestContainSubstring(t *testing.T) {
 				substring:  "world",
 				shouldFail: true,
 				errorCheck: func(t *testing.T, message string) {
-					if !strings.Contains(message, "Note: Case mismatch detected (use should.WithIgnoreCase() if intended)") {
-						t.Errorf("Expected message to contain note message, but got %q", message)
+					expectedParts := []string{
+						"Expected string to contain 'world', but found case difference",
+						"Substring: 'world'",
+						"Found    : 'WORLD' at position 7",
+						"Note: Case mismatch detected (use should.WithIgnoreCase() if intended)",
+					}
+					for _, part := range expectedParts {
+						if !strings.Contains(message, part) {
+							t.Errorf("Expected message to contain %q, but got:\n%s", part, message)
+						}
+					}
+				},
+			},
+			{
+				name:       "should show simplified case mismatch error for mixed case",
+				actual:     "Get \"http://127.0.0.1:56748\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)",
```

Comment:
Using string literal would be clearer

```suggestion
				actual:     `Get  "http://127.0.0.1:56748": context deadline exceeded (Client.Timeout exceeded while awaiting headers)`,
```